apps/prairielearn/assets/scripts/lib/accessRulesTable.ts

```diff
@@ -0,0 +1,153 @@
+import { html } from '@prairielearn/html';
+
+export function AccessRulesTable({
+  accessRules,
+  ptHost,
+  devMode,
+  hasCourseInstancePermissionView,
+  editMode,
+}: {
+  accessRules: any[];
+  ptHost: string;
+  devMode: boolean;
+  hasCourseInstancePermissionView: boolean;
+  editMode: boolean;
+}) {
+  return html`
+    <div
+      class="table-responsive js-access-rules-table"
+      id="table-responsive"
+      data-pt-host="${ptHost}"
+      data-dev-mode="${devMode}"
+      data-has-course-instance-permission-view="${hasCourseInstancePermissionView}"
+      data-timezone
+    >
+      <table class="table table-sm">
+        <thead>
+          <tr id="tableHeaderRow">
+            <th id="arrowButtonsHeader" ${editMode ? '' : 'hidden'}></th>
+            <th id="editButtonHeader" ${editMode ? '' : 'hidden'}></th>
+            <th id="deleteButtonHeader" ${editMode ? '' : 'hidden'}></th>
+            <th>Mode</th>
+            <th>UIDs</th>
+            <th>Start date</th>
+            <th>End date</th>
+            <th>Active</th>
+            <th>Credit</th>
+            <th>Time limit</th>
+            <th>Password</th>
+            <th>PrairieTest</th>
+          </tr>
+        </thead>
+        <tbody>
+          ${accessRules.map((access_rule, i) => {
+            // Only users with permission to view student data are allowed
+            // to see the list of uids associated with an access rule. Note,
+            // however, that any user with permission to view course code
+            // (or with access to the course git repository) will be able to
+            // see the list of uids, because these access rules are defined
+            // in course code. This should be changed in future, to protect
+            // student data. See https://github.com/PrairieLearn/PrairieLearn/issues/3342
+            return html`
+              <tr class="tableDataRow">
+                <td class="arrowButtonsCell align-content-center" ${editMode ? '' : 'hidden'}>
+                  <div>
+                    <button
+                      class="btn btn-xs btn-secondary up-arrow-button"
+                      type="button"
+                      data-row="${i}"
+                    >
+                      <i class="fa fa-arrow-up" aria-hidden="true"></i>
+                    </button>
+                  </div>
+                  <div>
+                    <button
+                      class="btn btn-xs btn-secondary down-arrow-button"
+                      type="button"
+                      data-row="${i}"
+                    >
+                      <i class="fa fa-arrow-down" aria-hidden="true"></i>
+                    </button>
```

Comment:
It'd be really nice to disable the up/down buttons for the first/last rules respectively!

Reply:
Done in [1dfef07](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/1dfef07527ed0296ff3cdfa7c58d85fce2304b97).